gallery_dl/extractor/steamgriddb.py

```diff
@@ -0,0 +1,211 @@
+# -*- coding: utf-8 -*-
+
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License version 2 as
+# published by the Free Software Foundation.
+
+"""Extractors for https://www.steamgriddb.com"""
+
+from .common import Extractor, Message
+from .. import text, exception
+
+
+BASE_PATTERN = r"(?:https?://)?(?:www\.)?steamgriddb\.com"
+LANGUAGE_CODES = (
+    "aa", "ab", "ae", "af", "ak", "am", "an", "ar", "as", "av", "ay", "az",
+    "ba", "be", "bg", "bh", "bi", "bm", "bn", "bo", "br", "bs", "ca", "ce",
+    "ch", "co", "cr", "cs", "cu", "cv", "cy", "da", "de", "dv", "dz", "ee",
+    "el", "en", "eo", "es", "et", "eu", "fa", "ff", "fi", "fj", "fo", "fr",
+    "fy", "ga", "gd", "gl", "gn", "gu", "gv", "ha", "he", "hi", "ho", "hr",
+    "ht", "hu", "hy", "hz", "ia", "id", "ie", "ig", "ii", "ik", "io", "is",
+    "it", "iu", "ja", "jv", "ka", "kg", "ki", "kj", "kk", "kl", "km", "kn",
+    "ko", "kr", "ks", "ku", "kv", "kw", "ky", "la", "lb", "lg", "li", "ln",
+    "lo", "lt", "lu", "lv", "mg", "mh", "mi", "mk", "ml", "mn", "mr", "ms",
+    "mt", "my", "na", "nb", "nd", "ne", "ng", "nl", "nn", "no", "nr", "nv",
+    "ny", "oc", "oj", "om", "or", "os", "pa", "pi", "pl", "ps", "pt", "qu",
+    "rm", "rn", "ro", "ru", "rw", "sa", "sc", "sd", "se", "sg", "si", "sk",
+    "sl", "sm", "sn", "so", "sq", "sr", "ss", "st", "su", "sv", "sw", "ta",
+    "te", "tg", "th", "ti", "tk", "tl", "tn", "to", "tr", "ts", "tt", "tw",
+    "ty", "ug", "uk", "ur", "uz", "ve", "vi", "vo", "wa", "wo", "xh", "yi",
+    "yo", "za", "zh", "zu",
+)
```

Comment:
This should ideally be added to the [CODES](https://github.com/mikf/gallery-dl/blob/master/gallery_dl/util.py#L467) list in util.py and used from there, but that sounds like a lot of effort. I guess this is fine as is.

Reply:
One could scrape https://en.wikipedia.org/w/index.php?title=List_of_ISO_639_language_codes, though we'd probably need to change `CODES` from being a `dict[str, str]` to `dict[str, list[str]]`

Reply:
Like I said: just leave it as is.
Or don't check for valid language codes.

Reply:
The server gives an unintuitive error if we don't:
```
> curl https://www.steamgriddb.com/api/public/search/assets -s -H "Content-Type: application/json" -H "Accept: application/json" --data '{"asset_type": "grid", "limit": 48, "game_id": [5259324], "languages": ["en"]}' | head -c50
{"success":true,"data":{"has_hidden_assets":true,"
> curl https://www.steamgriddb.com/api/public/search/assets -s -H "Content-Type: application/json" -H "Accept: application/json" --data '{"asset_type": "grid", "limit": 48, "game_id": [5259324], "languages": ["km"]}' | head -c50
{"success":true,"data":{"has_hidden_assets":false,
> curl https://www.steamgriddb.com/api/public/search/assets -s -H "Content-Type: application/json" -H "Accept: application/json" --data '{"asset_type": "grid", "limit": 48, "game_id": [5259324], "languages": ["pp"]}' | head -c50
{
    "message": "Slim Application Error"
}
```

We could perhaps add a message to it that says "This may be caused be invalid options", but filtering seems to be the easiest method